### ISSUE TITLE
[Feature] Add final tick number as third parameter to axis label formatter

### ISSUE THREAD

**Issue by JeffersonGarner** [apache/echarts#19794]
### What problem does this feature solve?

**Case**

Rather than show a numerical value of the last (upper bound) value on the Y Axis, it is sometimes desirable to show something else programmatically. 

**Context**

We had a goal to show a unit of measure (ex: kWh, m/s, or A) on the last Y axis label. Since assigning axis.splitNumber doesn't always linearly corelate to the number of ticks, we couldn't determine the last tick without risky code.

**Rationale**

By having the final tick number in the axis label formatter, we could determine what tick was the last one and easily write that logic within the formatter.  Without this, we could only entertain a solution [as described in Stack Overflow here](https://stackoverflow.com/a/77102205), which not intended to be done, and requires that the chart actually be rendered before we apply the actual formatting that we want.

### What does the proposed API look like?

**Current** ([described in docs](https://echarts.apache.org/en/option.html#xAxis.axisLabel.formatter))

`
formatter: function (value, index) {
    return value + 'kg';
}
`

**Proposed**

`
formatter: function (value, index, totalTicks) {
    const desiredLastTickLabel = "desired value here"
    const isLastTick = index === totalTicks - 1

    return isLastTick ? desiredLastTickLabel : value + 'kg';
}
`

**Comment by helgasoft** [comment]
> We had a goal to show a unit of measure (ex: kWh, m/s, or A) on the last Y axis label. 

Usually units of measurement are shown in axis' [name](https://echarts.apache.org/en/option.html#yAxis.name). 
But it can also be show on last label by using [max](https://echarts.apache.org/en/option.html#yAxis.max).
[Demo](https://echarts.apache.org/examples/en/editor.html?code=CYQwLiAEC8kNoEYBMAGANJBBmJGEoTywBYMBOdSJLSpfAXQG4AoAewAcwBLVgOxkgBvZpEjcwAGwCmALiEjRYqQA8wcgOQBPAILKuAZ3UKAvmgXLdBucMVjN7WZHUBjcFIDmrAE6b1Z26AQcnDqALJ8fk4AKgCuUpHqAOpSwAlRABYxCQBiXlwJAMrghTG86vQm_pA6evrWCqK8IAC2juoA1onpkU2tUSpgBWCa0jKCAGZ8YMlc7ulq6gBGrBKppg2QzSDKGoEgodt-G9sGADIgi1IS9baik15bYGBSXnLjpc7cfJAAFABuIAkcQwXF4wBUAEp5LdFPpxsoBACgVJoNADmB0gA6LbKH6Y_F7KEAfickE63UgGnULBhoi8UjAMS8_CRcUgAGpIHDlDTbsYNvzROtRPoXlwpHV4BsbLdhg4NBJQfEqgFwCA5HsBQoKsZGEA)

**Comment by JeffersonGarner** [comment]
> Usually units of measurement are shown in axis' [name](https://echarts.apache.org/en/option.html#yAxis.name).

Agree! We've got some big numbers and units of measures, so we want to save some space and provide more for the chart view.

> But it can also be show on last label by using [max](https://echarts.apache.org/en/option.html#yAxis.max).

I can see how this works, and is better than the [referenced workaround ](https://stackoverflow.com/questions/77096133/how-to-get-the-real-splitnumber-and-interval-of-a-axis-of-a-echarts-instance/77102205#77102205), but deriving the max from the data per axis does seem a bit inefficient.

**Comment by desjarlaisdumaishugo** [comment]
Hi,

This feature is exactly what I'm looking for!

Would this be easy to do in the echarts code ?

Thanks !

**Comment by MatthiasMert** [comment]
A possible workaround can be found [here](https://stackoverflow.com/questions/79047437/different-formatting-for-min-max-label-on-a-time-axis-with-echarts-js/79060790#79060790)
